aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -814,6 +814,10 @@ SparseTensor sparse_mask(const Tensor& t, const SparseTensor& mask) {
       " but mask has size ",
       mask.sizes());
 
+  if (t.is_same(mask)) {
+    return t;
+  }
```

Comment:
@cpuhrsch , @pearu , this is to make sure `x.sparse_mask(x)` is indeed efficient in forward for ops that, essentially, equivalent for dense and sparse in forward, but in backward require projected grads.